### PR TITLE
Adds BadEventHandlerError to EventMixin.

### PR DESCRIPTION
**Pull request by pathunstrom**
Regarding #106

### CODE REVIEW

File: ppb/events.py [hunk 1]
```diff
@@ -30,8 +33,26 @@ def __event__(self, bag, fire_event):
 
         meth = getattr(self, 'on_' + name, None)
         if callable(meth):
-            elog.debug(f"Calling handler {meth} for {name}")
-            meth(bag, fire_event)
+            try:
+                elog.debug(f"Calling handler {meth} for {name}")
+                meth(bag, fire_event)
+            except TypeError as ex:
```
[the remark this answers is not in the document]
Comment (pathunstrom) [reply ppb/events.py]: I am unsure of the best way to do such a thing.

I can parse the error I'm checking for wording.

Alternatively, I can do a simple check against the signature itself? Make sure it takes two arguments, and if it has type hinting make sure it had the correct types?

[the remark this answers is not in the document]
Comment (astronouth7303) [reply ppb/events.py]: I would do a check against the signature itself. You're calling it with two positional arguments. Make sure it has exactly 2 required arguments and both can be called positionally.

Alternatively, run this check in advance?

It's just that `TypeError` is kinda super common and you're totally swallowing exceptions from the event handler.

File: ppb/events.py [hunk 2]
```diff
@@ -22,16 +22,50 @@ def camel_to_snake(txt):
     return boundaries_finder_2.sub(r'\1_\2', s1).lower()
 
 
+class BadEventHandlerException(TypeError):
+
+    def __init__(self, object_type, method, event_type):
+        article = ['a', 'an'][int(type(event_type).__name__.lower()[0] in "aeiou")]
```
Comment (astronouth7303) [inline ppb/events.py]: Is that an extra `type()` or am I not thinking clearly?

Comment (pathunstrom) [reply ppb/events.py]: It totally is. I rearranged the code and forgot to get this one.

File: ppb/events.py [hunk 3]
```diff
@@ -22,16 +22,50 @@ def camel_to_snake(txt):
...
 class EventMixin:
     def __event__(self, bag, fire_event):
         elog = logging.getLogger('game.events')
 
         name = camel_to_snake(type(bag).__name__)
-
-        meth = getattr(self, 'on_' + name, None)
+        meth_name = 'on_' + name
+        meth = getattr(self, meth_name, None)
         if callable(meth):
```
[the remark this answers is not in the document]
Comment (pathunstrom) [reply ppb/events.py]: That's an extremely good question I'm going to need to ponder.

**Pull request merged**